### PR TITLE
code changes for data validation record count from hive and hdfs and code changes in hive-lib

### PR DESCRIPTION
**Pull request by ritaliu912**

### CODE REVIEW

File: bigdime-validators/src/main/java/io/bigdime/validation/RecordCountFromHiveValidator.java [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+ * record count mismatches --- validation fail
+ * otherwise, validation success
+ * 
+ * @author Rita Liu
+ */
+
+@Factory(id = "record_count_hive", type = RecordCountFromHiveValidator.class)
+@Component
+@Scope("prototype")
+public class RecordCountFromHiveValidator implements Validator {
```
Comment (a-ninja) [inline bigdime-validators/src/main/java/io/bigdime/validation/RecordCountFromHiveValidator.java]: I'd suggest renaming this to HiveRecordCountValidator.

File: bigdime-validators/src/main/java/io/bigdime/validation/RecordCountFromHiveValidator.java [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+	protected boolean isReadyToValidate(final ActionEvent actionEvent) {
+		String hdfsBasePath = actionEvent.getHeaders().get(ActionEventHeaderConstants.HDFS_PATH);
+		String hdfsFileName = actionEvent.getHeaders().get(ActionEventHeaderConstants.HDFS_FILE_NAME);
+
+
+		String totalSize = actionEvent.getHeaders().get(ActionEventHeaderConstants.SOURCE_FILE_TOTAL_SIZE);
+		String totalRead = actionEvent.getHeaders().get(ActionEventHeaderConstants.SOURCE_FILE_TOTAL_READ);
+		
+		String validationReady = actionEvent.getHeaders().get(ActionEventHeaderConstants.VALIDATION_READY);
+		if (validationReady == null || !validationReady.equalsIgnoreCase("true")) {
```
Comment (a-ninja) [inline bigdime-validators/src/main/java/io/bigdime/validation/RecordCountFromHiveValidator.java]: So, VALIDATION_READY must be not null and set to true for validator to execute?

**Comment by ritaliu912** [comment]
will reopen it

**Comment by ritaliu912** [comment]
reopen

**Comment by mnamburi** [comment]
looks good to me

**Pull request merged**